gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/diagnosticreport/SpecimenCompoundsMapper.java

```diff
@@ -107,6 +115,31 @@ public List<Observation> handleSpecimenChildComponents(RCMRMT030101UKEhrExtract
         return batteryObservations;
     }
 
+    private static @NotNull List<RCMRMT030101UKCompoundStatement> getNestedSpecimenCompoundStatements(
+        RCMRMT030101UKCompoundStatement specimenCompoundStatement
+    ) {
+        return specimenCompoundStatement.getComponent()
+            .stream()
+            .filter(RCMRMT030101UKComponent02::hasCompoundStatement)
+            .map(RCMRMT030101UKComponent02::getCompoundStatement)
+            .filter(SpecimenCompoundsMapper::hasBatteryOrClusterClassCode)
+            .toList();
+    }
+
+    private void handleSpecimenObservationStatement(
+        List<Observation> observations,
+        DiagnosticReport diagnosticReport,
+        RCMRMT030101UKCompoundStatement specimenCompoundStatement
+    ) {
+        for (var specimenObservationStatement : getObservationStatementsInCompound(specimenCompoundStatement)) {
+            getObservationById(observations, specimenObservationStatement.getId().getRoot())
+                .ifPresent(observation -> {
+                    handleObservationStatement(specimenCompoundStatement, specimenObservationStatement, observation);
+                    DiagnosticReportMapper.addResultToDiagnosticReport(observation, diagnosticReport);
+                });
+        }
+    }
```

Comment:
Not too fussed about this one at all, just noticed that a lot of the mappers are utilising a more functional approach, so i was wondering if it would be worth doing something like:

```suggestion
    private void handleSpecimenObservationStatement(
        List<Observation> observations,
        DiagnosticReport diagnosticReport,
        RCMRMT030101UKCompoundStatement specimenCompoundStatement
    ) {
        getObservationStatementsInCompound(specimenCompoundStatement)
            .forEach(specimenObservationStatement -> {
                getObservationById(observations, specimenObservationStatement.getId().getRoot())
                    .ifPresent(observation -> {
                        handleObservationStatement(specimenCompoundStatement, specimenObservationStatement, observation);
                        DiagnosticReportMapper.addResultToDiagnosticReport(observation, diagnosticReport);
                    });
            });
    }
```

Reply:
I thought the foreach was marginally less easy to read, but i'm open to either way